src/twisted/web/_newclient.py

```diff
@@ -1568,7 +1576,7 @@ def _disconnectParser(self, reason):
             # transport.  Stop proxying from the parser's transport to the real
             # transport before telling the parser it's done so that it can't do
             # anything.
-            self._transportProxy._stopProxying()
+            self._transportProxy.stopProxying()
```

Comment:
Beyond just making the method public, is there much value in the method at all?  The producer is set to None immediately after it is called, is it common for something else to keep a reference?  Aside from that question making it public does seem sane, though somewhat unrelated to the problem at hand.

Reply:
If you don't have stopProxying, you will have to access the private `self._transportProxy._producer.stopProxying()`

I think that there is a lot of value in this method as it encapsulated a well-defined action on the transport.

For now we are only (conditionally) stop producing, but in the future there might be other internal actions to be done for stopProxying

With a public method, the other users don't have to care about _producer and whether it is None or not.